packages/drivekit_trip_analysis/flutter_drivekit_trip_analysis_android/android/src/main/kotlin/com/drivequant/drivekit/flutter/tripanalysis/init/TripNotificationInitializer.kt

```diff
@@ -0,0 +1,109 @@
+        val appName = context.applicationContext.packageName
+
+        val notification = buildNotification(
+            title = resources.getIdentifier(ResourceId.TITLE, STRING, appName, LogStatusWhenNotFound.ERROR).let { id ->
+                getWithDefaultValueIfNotFound(id, { resources.getString(id) }, "DriveKit")
+            },
+            content = resources.getIdentifier(ResourceId.CONTENT, STRING, appName, LogStatusWhenNotFound.ERROR).let { id ->
+                getWithDefaultValueIfNotFound(id, { resources.getString(id) }, "DriveKit is running")
+            },
+            iconDrawableId = resources.getIdentifier(ResourceId.ICON, DRAWABLE, appName, LogStatusWhenNotFound.ERROR),
+            cancelIconDrawableId = resources.getIdentifier(ResourceId.CANCEL_ICON, DRAWABLE, appName, LogStatusWhenNotFound.WARNING).let {
+                getWithDefaultValueIfNotFound(it, { it }, null)
+            },
+            notificationId = resources.getIdentifier(ResourceId.NOTIF_ID, INTEGER, appName, LogStatusWhenNotFound.WARNING).let {
+                getWithDefaultValueIfNotFound(it, { resources.getInteger(it) }, null)
+            },
+            enableCancelValue = resources.getIdentifier(ResourceId.ENABLE_CANCEL, BOOL, appName, LogStatusWhenNotFound.WARNING).let { id ->
+                getWithDefaultValueIfNotFound(id, { resources.getBoolean(id) }, false)
+            },
+            cancelText = resources.getIdentifier(ResourceId.CANCEL_CONTENT, STRING, appName, LogStatusWhenNotFound.WARNING).let { id ->
+                getWithDefaultValueIfNotFound(id, { resources.getString(id) }, null)
+            },
+            context = context
+        )
+
+        DriveKitTripAnalysis.tripNotification = notification
+        return notification
+    }
+
+    private fun <T> getWithDefaultValueIfNotFound(id: Int, provider: () -> T, defaultValue: T) = if (id != 0) provider() else defaultValue
+
+    private fun buildNotification(
+        title: String,
+        content: String,
+        iconDrawableId: Int,
+        enableCancelValue: Boolean,
+        cancelIconDrawableId: Int?,
+        cancelText: String?,
+        notificationId: Int?,
+        context: Context
+    ) = TripNotification(
+        title = title,
+        content = content,
+        iconId = iconDrawableId
+    ).apply {
+        this.notificationId = notificationId // TODO create a setter in flutter to change it at exec time
+        enableCancel = enableCancelValue
+        cancel = cancelText
+        cancelIconId = cancelIconDrawableId
+        channelId = title
+        val intent = context.packageManager.getLaunchIntentForPackage(context.applicationContext.packageName)
+        contentIntent = TaskStackBuilder.create(context)
+            .addNextIntentWithParentStack(intent)
+            .getPendingIntent(0, PendingIntent.FLAG_IMMUTABLE)
+    }
+
+    override fun dependencies(): List<Class<out Initializer<*>>> {
+        // No dependencies on other libraries.
```

Comment:
Je ferais une dépendance au startup qui est présent dans Core. Voir [exemple](https://github.com/DriveQuantPublic/drivekit-ui-android/blob/d685aebd6b78f8c85e334d102ddf89779976f720/TripAnalysisUI/src/main/java/com/drivequant/drivekit/tripanalysis/DriveKitTripAnalysisUIInitializer.kt#L16) dans nos modules UI qui utilisent également Startup.

```suggestion
        return listOf(DriveKitInitializer::class.java)
```

(avec l'import `import com.drivequant.drivekit.core.autoinit.DriveKitInitializer`)

Reply:
Ah super ça !